lib/ui/process_ui/widgets/date_control.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+import 'package:provider/provider.dart';
+import 'package:registration_client/model/field.dart';
+import 'package:registration_client/provider/global_provider.dart';
+import 'package:registration_client/provider/registration_task_provider.dart';
+import 'package:registration_client/ui/process_ui/widgets/custom_label.dart';
+import 'package:registration_client/utils/app_style.dart';
+
+
+class DateControl extends StatefulWidget {
+  const DateControl({super.key, required this.field, required this.validation});
+  final Field  field;
+  final RegExp validation;
+
+  @override
+  State<DateControl> createState() => _DateControlState();
+}
+
+class _DateControlState extends State<DateControl> {
+  bool isMvelValid = true;
+  TextEditingController dateController = TextEditingController();
+  DateTime? pickedDate;
+
+  @override
+  void initState() {
+    _getDataFromMap();
+    super.initState();
+  }
+
+  void _selectDate() async {
+    pickedDate = await showDatePicker(
+        context: context,
+        initialDate: DateTime.now(),
+        firstDate: DateTime(
+            2000),
+        lastDate: DateTime(2101));
```

Comment:
any reason we use last date and first date as these specific values

Reply:
Not given proper validation so changed last date with current year and first date with 100 yrs diff from curr date

Reply:
ok let's check what's needed here.